### PR TITLE
The Infiltrator Sneaksuit Bundle: Armor for (relatively) Cheap

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Redoing the pull request because it's changed so fucking much. Original over here #49550.

The Infiltrator Bundle is a sneaksuit set that allows the wearer the benefits of a decent armor set at a relatively low price of 6TC. The armor vest paired with the sneaksuit will equal a syndicate hardsuit, but only on the chest. It also carries none of the benefits of the hardsuit, such as EVA. 

The bundle comes complete with the following items:

**Infiltrator Helmet:** It's a insidious helmet. Has flashproofing.
**Infiltrator Vest:** It's roughly equal to an armor vest, but a lot wearker against lasers and disablers.
**Blood-red Sneaksuit:** It's the highest armored undersuit in the game, at 10 across the board.
**Infiltrator Balaclava:** Equal to an explorer mask in armor, and has a diagnostic hud built in. It also distorts your voice, so you always come up as 'Unknown' when speaking, even when you have another ID on. Obviously has anti-syngery with the Agent Card, but you're not wearing this to disguise as other people.
**Infiltrator Gloves:** Special combat gloves that allow you to fireman carry people more quickly.
**Sneakboots:** They muffle your footsteps completely!
**Carrying Case:** Storage container for the full suit and guns if you have them. Guns not included.

Screenshot 2020-02-28 06 08 03

ss+(2020-02-26+at+08 29 43)

Also includes the Sleepytime Pajama Bundle for nukies.

Zawo and I have been passing this between us for a while trying to get this into a good state. Hopefully now it's worth an inclusion.

## Why It's Good For The Game

~~A cheaper alternative to the Syndicate Balloon to indicate to everyone around you that you are probably valid.~~

A recognizable outfit that proves to be useful in minor ways and provide cheap armor for traitors. Nothing this grants is especially powerful on it's own, but it is a useful for traitors wanting to gain an edge at low cost.

## Changelog
:cl: NecromancerAnne and zawo
add: The Infiltrator Bundle, an armor kit for 6TC. Murder people in style!
add: Some pajamas for nukies to get plenty of bed rest.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
As I asked in the other PR, should this be available to nukies y/n?

**Comment by Pumpkinoe** [comment]
Love the look of it, and what the playstyle seems to encourage. Are the infiltrator gloves insulated like regular combat gloves? If not, that seems counter-intuitive for the whole 'I'm a hackerman' thing. Also please call the carrying case the 'Syndicase'

**Comment by Pumpkinoe** [comment]
Also, is the carrying case able to hold guns like shotguns, sniper rifles or other longarm ballistic?

**Comment by shellspeed1** [comment]
Good night Disky

**Comment by Tharcoonvagh** [comment]
Sure

File: code/modules/clothing/masks/boxing.dm [hunk 1]
```diff
@@ -11,6 +11,38 @@
...
+	
+	var/vunknown = FALSE ///This makes it so that your name shows up as unknown when wearing the mask.
+
+/obj/item/clothing/mask/infiltrator/equipped(mob/living/carbon/human/user, slot)
+	..()
+	if(slot != ITEM_SLOT_MASK)
+		return
+	to_chat(user, "You roll the balaclava over your face, and a data display appears before your eyes.")
+	ADD_TRAIT(user, TRAIT_DIAGNOSTIC_HUD, MASK_TRAIT)
+	var/datum/atom_hud/H = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
```
Comment (GunHog) [inline code/modules/clothing/masks/boxing.dm]: You may wish to use the BASIC version here. ADVANCED is the silicon version, and the key difference is that ADVANCED shows bot routes (such as Beepsky's path to the next patrol beacon). Human players would probably consider it clutter, so I do not recommend using it.

Comment (necromanceranne) [reply code/modules/clothing/masks/boxing.dm]: noted, cheers

File: code/datums/components/footstep.dm [hunk 2]
```diff
@@ -94,11 +94,15 @@
 	var/feetCover = (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET))
 
 	if(H.shoes || feetCover) //are we wearing shoes
+		if(HAS_TRAIT(H, TRAIT_SILENT_FOOTSTEPS))
```
Comment (Rohesie) [inline code/datums/components/footstep.dm]: You could make these two checks as single one at the start of the proc instead.
Something like:
```
/datum/component/footstep/proc/play_humanstep()
  if(HAS_TRAIT(parent, TRAIT_SILENT_FOOTSTEPS))
    return
```

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 3]
```diff
@@ -18,6 +18,24 @@
...
+	icon_state = "sneakboots"
+	item_state = "sneakboots"
+	resistance_flags = FIRE_PROOF |  ACID_PROOF
+
+/obj/item/clothing/shoes/combat/sneakboots/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(slot != ITEM_SLOT_FEET)
+		return
+	if(slot == ITEM_SLOT_FEET)
+		ADD_TRAIT(user, TRAIT_SILENT_FOOTSTEPS, SHOES_TRAIT)
```
Comment (Rohesie) [inline code/modules/clothing/shoes/miscellaneous.dm]: If the first check passes, the second will necessarily be truthy. You can remove one of them.

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 4]
```diff
@@ -18,6 +18,24 @@
...
+
+/obj/item/clothing/shoes/combat/sneakboots/dropped(mob/living/carbon/human/user)
+	. = ..()
+	REMOVE_TRAIT(user, TRAIT_SILENT_FOOTSTEPS, SHOES_TRAIT)
```
Comment (Rohesie) [inline code/modules/clothing/shoes/miscellaneous.dm]: You might want to perform the action first and later call parent with `return ..()` on removal/deleting procs in general, as parent calls may do some cleanup actions changing the state of the item.
As in, replace `. = ..()` in the beginning with `return ..()` at the end

File: code/modules/clothing/suits/armor.dm [hunk 5]
```diff
@@ -172,6 +172,15 @@
...
 	allowed = GLOB.detective_vest_allowed
 
+/obj/item/clothing/suit/armor/vest/infiltrator
+	name = "infiltrator vest"
+	desc = "This vest appears to be made of of highly flexible materials that absorb impacts with ease."
+	icon_state = "infiltrator"
+	item_state = "infiltrator"
+	armor = list("melee" = 30, "bullet" = 40, "laser" = 20, "energy" = 30, "bomb" = 70, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	strip_delay = 80
```
Comment (Rohesie) [inline code/modules/clothing/suits/armor.dm]: I'm personally fan of time defines. `strip_delay = 8 SECONDS` looks much clearer to me.

File: code/modules/mob/living/carbon/human/say.dm [hunk 6]
```diff
@@ -16,6 +16,10 @@
 				return real_name
 		else
 			return real_name
+	if(istype(wear_mask, /obj/item/clothing/mask/infiltrator))
+		var/obj/item/clothing/mask/infiltrator/V = wear_mask
+		if(V.vunknown)
+			return ("Unknown")
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/say.dm]: This is a property pretty similar to the chameleon's `vchange` and could be generalized as a mask property.
As in moving both variables to the mask level (maybe renaming it to something more clear like `voice_change` and `voice_unknown`) and performing a single `istype` check here. More generic behavior.

**Comment by Ziiro** [comment]
Honest to god I'll probably buy this just because it looks so good.

**Comment by Pumpkinoe** [comment]
> Honest to god I'll probably buy this just because it looks so good.

same, it looks sexy as hell
also, i like the case just as much, and the concept of using the syndicate holster + silenced stechkin just sounds like a bunch of fun

**Comment by itseasytosee** [comment]
Mutch better than OG PR Love it

**Comment by Pumpkinoe** [comment]
Maybe mention somehow about the 15 damage of the case?

**Comment by Spepbuthewranglesgithub** [comment]
on one hand it is infact very cool, but on the other there is a immense lack of sneak to it, i give this a 4/10.

**Comment by necromanceranne** [comment]
You would be surprised how spooky it is when your footsteps make no sound. Once you had gotten used to them as an auditory cue, the removal makes it a bit surprising when someone approaches you without them in the dark.

**Comment by Pumpkinoe** [comment]
Maybe to encourage stealth in the dark it should come with night vision? just throwing that out there

File: code/modules/clothing/masks/boxing.dm [hunk 7]
```diff
@@ -11,6 +11,38 @@
...
+	flags_inv = HIDEFACE|HIDEHAIR|HIDEFACIALHAIR
+	visor_flags_inv = HIDEFACE|HIDEFACIALHAIR
+	w_class = WEIGHT_CLASS_SMALL
+	armor = list("melee" = 10, "bullet" = 5, "laser" = 5,"energy" = 5, "bomb" = 0, "bio" = 0, "rad" = 10, "fire" = 100, "acid" = 40)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	
+	var/voice_unknown = FALSE ///This makes it so that your name shows up as unknown when wearing the mask.
+
+/obj/item/clothing/mask/infiltrator/equipped(mob/living/carbon/human/user, slot)
+	..()
```
Comment (Rohesie) [inline code/modules/clothing/masks/boxing.dm]:
```suggestion
	. = ..()

```

File: code/modules/clothing/masks/boxing.dm [hunk 8]
```diff
@@ -11,6 +11,38 @@
...
+	if(slot != ITEM_SLOT_MASK)
+		return
+	to_chat(user, "You roll the balaclava over your face, and a data display appears before your eyes.")
+	ADD_TRAIT(user, TRAIT_DIAGNOSTIC_HUD, MASK_TRAIT)
+	var/datum/atom_hud/H = GLOB.huds[DATA_HUD_DIAGNOSTIC_BASIC]
+	H.add_hud_to(user)
+	voice_unknown = TRUE
+
+/obj/item/clothing/mask/infiltrator/dropped(mob/living/carbon/human/user)
+	..()
```
Comment (Rohesie) [inline code/modules/clothing/masks/boxing.dm]:
```suggestion
```

File: code/modules/clothing/masks/boxing.dm [hunk 9]
```diff
@@ -11,6 +11,38 @@
...
+	to_chat(user, "You pull off the balaclava, and the mask's internal hud system switches off quietly.")
+	REMOVE_TRAIT(user, TRAIT_DIAGNOSTIC_HUD, MASK_TRAIT)
+	var/datum/atom_hud/H = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	H.remove_hud_from(user)
+	voice_unknown = FALSE
```
Comment (Rohesie) [inline code/modules/clothing/masks/boxing.dm]:
```suggestion
	voice_unknown = FALSE
	return ..()
```

**Comment by Coastfront** [comment]
I really like this, it looks fucking beautiful is the first major thing, And honestly the whole thing just seems very robust for a spooky evil man TM gimmick. And seems to have pretty good synergy with some other traitor items.10/10 thing honestly.

**Comment by Kerbin-Fiber** [comment]
yes very cool speedmerge plzkthx

**Comment by optimumtact** [comment]
3tc is just too cheap for flash immunity, armour, identity hiding and no footsteps

**Comment by necromanceranne** [comment]
If you feel it's a little cheap, sure. Would it being 4-5 tc be okay? Worth noting, it doesn't have the advantage of EVA, and the syndicate softsuit is still a suit of thick material armor in it's own right that comes with flashproofing on the helmet. 

I felt like EVA was something well and truly valuable in it's own right even with slowdown (since there are ways to get around that).

**Comment by optimumtact** [comment]
I think you could just squeak by on 4, and anything above that is fine because it can be adjusted down easily if needed.

**Comment by necromanceranne** [comment]
6 TC should be a safe range for now. If the armor is 4TC, the gloves and case are 1TC (comparing to the syndicate toolbox), and the extra special gear another 1TC, then 6 seems more than fair.

**Pull request merged**